### PR TITLE
Changes related mostly with support for ISARA SampleChanger and SampleChangerMaintenance features for mxcube3

### PR DESCRIPTION
**Pull request by vrey01**
Includes different small fixes commits plus

- TangoLimaVideoDevice - independent of Qt (use opencv instead for image encoding)
- SampleChanger (ISARA) and changes for mxcube3

@meguiraun

### CODE REVIEW

File: HardwareObjects/GenericVideoDevice.py [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+
+            if self.cam_mirror is not None:
+                qimage = qimage.mirrored(self.cam_mirror[0], self.cam_mirror[1])     
+
+            qpixmap = QPixmap(qimage)
+            self.emit("imageReceived", qpixmap)
+            return qimage
+
+    def get_jpg_image(self):
+        """ for now this function allows to deal with prosilica or any RGB encoded video data"""
```
Comment (marcus-oscarsson) [inline HardwareObjects/GenericVideoDevice.py]: Pay attention to double comment, if not made attentionaly ?

File: HardwareObjects/GenericVideoDevice.py [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+            
+        else:
+            if bw:
+                return qimage.convertToFormat(QImage.Format_Mono)
+            else:
+                return qimage
+
+    def get_scaling_factor(self):
+        """
+        Descript. :
```
Comment (marcus-oscarsson) [inline HardwareObjects/GenericVideoDevice.py]: Descript. does not mean anything in sphinx (at least not to my knowledge),  I would suggest to remove it or replace it with something that tells us what the method does. As far as I know returns should also be :returns: .... if we are not using some alternative sphinx syntax ?

File: HardwareObjects/MAXIV/MaxIVSession.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+"""
```
Comment (marcus-oscarsson) [inline HardwareObjects/MAXIV/MaxIVSession.py]: Well done, this is exactly how Session was meant to be used. Thanks !!!

**Comment by marcus-oscarsson** [comment]
Hey !

Thanks for the nice work, do you think you could take a look at the comments ? Ill try to take another look next week and merge 

Marcus

**Comment by mguijarr** [comment]
Hello :)

It would be much easier for me if the PR could be split into several smaller ones, one for each thing it is modifying/adding instead of a bunch of commits.

Thanks in advance !

**Comment by meguiraun** [comment]
hej @mguijarr, is there any commit in particular that you have trouble with? So I can reduce the number of PRs, e.g. single PR for the ok commits and another one for the one that you might want to discuss in detail.

**Comment by mguijarr** [comment]
The PRs should be grouped by feature or by bug fix as much as possible, as changes do not only concern site specific files. Otherwise it makes review much more difficult.

**Comment by meguiraun** [comment]
see https://github.com/mxcube/HardwareRepository/pull/48

**Pull request closed without merging**